op-e2e/actions/upgrades/utils/assertions.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"math/big"
+
+	actionhelpers "github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/interop/dsl"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/lmittmann/w3"
+	"github.com/stretchr/testify/require"
+)
+
+var ProxyImplGetterFunc = w3.MustNewFunc(`implementation()`, `address`)
+
+func RequireUpgradeSuccessful(t actionhelpers.Testing, actors *dsl.InteropActors, implAddr common.Address, proxyAddress common.Address, activationBlockNumber *big.Int) {
```

Comment:
I wonder if there is a more descriptive name for this. Like `RequireProxiedContractExists`?